Crash using `isinstance` + `super` due to Astroid

### Bug description

When parsing the following ``a.py``:

```python
class A:
    def __init__(self):
        if not isinstance(super(), float):
            pass

```

there is a crash. Note however that this snippet is fine:

```python
class A:
    def __init__(self):
        if isinstance(super(), float):
            pass

```
with `not` removed. Also I believe it's nothing to do with `float`: you can replace this with a custom class, or another built-in.


### Command used

```shell
pylint a.py
```

### Pylint output

<details open>
    <summary>
        pylint crashed with a ``AstroidError`` and with the following stacktrace:
    </summary>

```python
Traceback (most recent call last):
  File "/home/bugreport/.venv/lib/python3.10/site-packages/astroid/inference_tip.py", line 33, in _inference_tip_cached
    result = _cache[func, node]
KeyError: (<function register_builtin_transform.<locals>._transform_wrapper at 0x7fbe7b0e5b40>, <Call l.3 at 0x7fbe7a87c910>)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 772, in _lint_file
    check_astroid_module(module)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1001, in check_astroid_module
    retval = self._check_astroid_module(
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1047, in _check_astroid_module
    walker.walk(node)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  [Previous line repeated 1 more time]
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/checkers/typecheck.py", line 1903, in visit_unaryop
    for error in node.type_errors():
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/nodes/node_classes.py", line 4466, in type_errors
    return [
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/nodes/node_classes.py", line 4466, in <listcomp>
    return [
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/inference.py", line 565, in _infer_unaryop
    for operand in self.operand.infer(context):
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/nodes/node_ng.py", line 161, in infer
    results = list(self._explicit_inference(self, context, **kwargs))
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/inference_tip.py", line 40, in _inference_tip_cached
    result = _cache[func, node] = list(func(*args, **kwargs))
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/brain/brain_builtin_inference.py", line 175, in _transform_wrapper
    result = transform(node, context=context)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/brain/brain_builtin_inference.py", line 725, in infer_isinstance
    isinstance_bool = helpers.object_isinstance(obj_node, class_container, context)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/helpers.py", line 133, in object_isinstance
    obj_type = object_type(node, context)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/helpers.py", line 87, in object_type
    types = set(_object_type(node, context))
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/astroid/helpers.py", line 71, in _object_type
    raise AssertionError(f"We don't handle {type(inferred)} currently")
AssertionError: We don't handle <class 'astroid.objects.Super'> currently

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 736, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/home/namc/pylintbugreport/.venv/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 774, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError

```


</details>

### Expected behaviour

No crash.

### Pylint version

```shell
pylint 3.0.0a6
astroid 2.15.1
Python 3.10.8 (main, Dec 19 2022, 16:27:32) [GCC 9.4.0]
```

### OS / Environment

linux (Linux)

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -547,6 +547,7 @@ contributors:
 - Arun Persaud <arun@nubati.net>
 - Arthur Lutz <arthur.lutz@logilab.fr>
 - Antonio Ossa <aaossa@uc.cl>
+- Antonio <antonio@zoftko.com>
 - Anthony VEREZ <anthony.verez.external@cassidian.com>
 - Anthony Tan <tanant@users.noreply.github.com>
 - Anthony Foglia <afoglia@users.noreply.github.com> (Google): Added simple string slots check.

--- a/doc/user_guide/configuration/all-options.rst
+++ b/doc/user_guide/configuration/all-options.rst
@@ -233,7 +233,7 @@ Standard Checkers
 
    confidence = ["HIGH", "CONTROL_FLOW", "INFERENCE", "INFERENCE_FAILURE", "UNDEFINED"]
 
-   disable = ["raw-checker-failed", "bad-inline-option", "locally-disabled", "file-ignored", "suppressed-message", "useless-suppression", "deprecated-pragma", "use-symbolic-message-instead", "consider-using-augmented-assign"]
+   disable = ["consider-using-augmented-assign"]
 
    enable = []
 

--- a/doc/whatsnew/2/2.17/index.rst
+++ b/doc/whatsnew/2/2.17/index.rst
@@ -29,6 +29,81 @@ so we find problems before the actual release.
 
 .. towncrier release notes start
 
+What's new in Pylint 2.17.5?
+----------------------------
+Release date: 2023-07-26
+
+
+False Positives Fixed
+---------------------
+
+- Fix a false positive for ``unused-variable`` when there is an import in a
+  ``if TYPE_CHECKING:`` block and ``allow-global-unused-variables`` is set to
+  ``no`` in the configuration.
+
+  Closes #8696 (`#8696 <https://github.com/pylint-dev/pylint/issues/8696>`_)
+
+- Fix false positives generated when supplying arguments as ``**kwargs`` to IO
+  calls like open().
+
+  Closes #8719 (`#8719 <https://github.com/pylint-dev/pylint/issues/8719>`_)
+
+- Fix a false positive where pylint was ignoring method calls annotated as
+  ``NoReturn`` during the ``inconsistent-return-statements`` check.
+
+  Closes #8747 (`#8747 <https://github.com/pylint-dev/pylint/issues/8747>`_)
+
+- Exempt parents with only type annotations from the ``invalid-enum-extension``
+  message.
+
+  Closes #8830 (`#8830 <https://github.com/pylint-dev/pylint/issues/8830>`_)
+
+
+
+Other Bug Fixes
+---------------
+
+- Fixed crash when a call to ``super()`` was placed after an operator (e.g.
+  ``not``).
+
+  Closes #8554 (`#8554 <https://github.com/pylint-dev/pylint/issues/8554>`_)
+
+- Fix crash for ``modified-while-iterating`` checker when deleting
+  members of a dict returned from a call.
+
+  Closes #8598 (`#8598 <https://github.com/pylint-dev/pylint/issues/8598>`_)
+
+- Fix crash in ``invalid-metaclass`` check when a metaclass had duplicate
+  bases.
+
+  Closes #8698 (`#8698 <https://github.com/pylint-dev/pylint/issues/8698>`_)
+
+- Avoid ``consider-using-f-string`` on modulos with brackets in template.
+
+  Closes #8720. (`#8720 <https://github.com/pylint-dev/pylint/issues/8720>`_)
+
+- Fix a crash when ``__all__`` exists but cannot be inferred.
+
+  Closes #8740 (`#8740 <https://github.com/pylint-dev/pylint/issues/8740>`_)
+
+- Fix crash when a variable is assigned to a class attribute of identical name.
+
+  Closes #8754 (`#8754 <https://github.com/pylint-dev/pylint/issues/8754>`_)
+
+- Fixed a crash when calling ``copy.copy()`` without arguments.
+
+  Closes #8774 (`#8774 <https://github.com/pylint-dev/pylint/issues/8774>`_)
+
+
+
+Other Changes
+-------------
+
+- Fix a crash when a ``nonlocal`` is defined at module-level.
+
+  Closes #8735 (`#8735 <https://github.com/pylint-dev/pylint/issues/8735>`_)
+
+
 What's new in Pylint 2.17.4?
 ----------------------------
 Release date: 2023-05-06

--- a/doc/whatsnew/fragments/8554.bugfix
+++ b/doc/whatsnew/fragments/8554.bugfix
@@ -1,3 +1,0 @@
-Fixed crash when a call to ``super()`` was placed after an operator (e.g. ``not``).
-
-Closes #8554

--- a/doc/whatsnew/fragments/8598.bugfix
+++ b/doc/whatsnew/fragments/8598.bugfix
@@ -1,4 +1,0 @@
-Fix crash for ``modified-while-iterating`` checker when deleting
-members of a dict returned from a call.
-
-Closes #8598

--- a/doc/whatsnew/fragments/8696.false_positive
+++ b/doc/whatsnew/fragments/8696.false_positive
@@ -1,3 +1,0 @@
-Fix a false positive for ``unused-variable`` when there is an import in a ``if TYPE_CHECKING:`` block and ``allow-global-unused-variables`` is set to ``no`` in the configuration.
-
-Closes #8696

--- a/doc/whatsnew/fragments/8698.bugfix
+++ b/doc/whatsnew/fragments/8698.bugfix
@@ -1,3 +1,0 @@
-Fix crash in ``invalid-metaclass`` check when a metaclass had duplicate bases.
-
-Closes #8698

--- a/doc/whatsnew/fragments/8719.false_positive
+++ b/doc/whatsnew/fragments/8719.false_positive
@@ -1,3 +1,0 @@
-Fix false positives generated when supplying arguments as ``**kwargs`` to IO calls like open().
-
-Closes #8719

--- a/doc/whatsnew/fragments/8720.bugfix
+++ b/doc/whatsnew/fragments/8720.bugfix
@@ -1,3 +1,0 @@
-Avoid ``consider-using-f-string`` on modulos with brackets in template.
-
-Closes #8720.

--- a/doc/whatsnew/fragments/8735.other
+++ b/doc/whatsnew/fragments/8735.other
@@ -1,3 +1,0 @@
-Fix a crash when a ``nonlocal`` is defined at module-level.
-
-Closes #8735

--- a/doc/whatsnew/fragments/8740.bugfix
+++ b/doc/whatsnew/fragments/8740.bugfix
@@ -1,3 +1,0 @@
-Fix a crash when ``__all__`` exists but cannot be inferred.
-
-Closes #8740

--- a/doc/whatsnew/fragments/8747.false_positive
+++ b/doc/whatsnew/fragments/8747.false_positive
@@ -1,3 +1,0 @@
-Fix a false positive where pylint was ignoring method calls annotated as ``NoReturn`` during the ``inconsistent-return-statements`` check.
-
-Closes #8747

--- a/doc/whatsnew/fragments/8754.bugfix
+++ b/doc/whatsnew/fragments/8754.bugfix
@@ -1,3 +1,0 @@
-Fix crash when a variable is assigned to a class attribute of identical name.
-
-Closes #8754

--- a/doc/whatsnew/fragments/8774.bugfix
+++ b/doc/whatsnew/fragments/8774.bugfix
@@ -1,3 +1,0 @@
-Fixed a crash when calling ``copy.copy()`` without arguments.
-
-Closes #8774

--- a/doc/whatsnew/fragments/8830.false_positive
+++ b/doc/whatsnew/fragments/8830.false_positive
@@ -1,4 +1,0 @@
-Exempt parents with only type annotations from the ``invalid-enum-extension``
-message.
-
-Closes #8830

--- a/pylint/__pkginfo__.py
+++ b/pylint/__pkginfo__.py
@@ -9,7 +9,7 @@ It's updated via tbump, do not modify.
 
 from __future__ import annotations
 
-__version__ = "2.17.4"
+__version__ = "2.17.5"
 
 
 def get_numversion_from_version(v: str) -> tuple[int, int, int]:

--- a/tbump.toml
+++ b/tbump.toml
@@ -1,7 +1,7 @@
 github_url = "https://github.com/pylint-dev/pylint"
 
 [version]
-current = "2.17.4"
+current = "2.17.5"
 regex = '''
 ^(?P<major>0|[1-9]\d*)
 \.

--- a/towncrier.toml
+++ b/towncrier.toml
@@ -1,5 +1,5 @@
 [tool.towncrier]
-version = "2.17.4"
+version = "2.17.5"
 directory = "doc/whatsnew/fragments"
 filename = "doc/whatsnew/2/2.17/index.rst"
 template = "doc/whatsnew/fragments/_template.rst"

